src/lib/race.ts

```diff
@@ -242,3 +244,171 @@ export async function race(
+  validateBooleanOpt('highlightBars', options.highlightBars);
+  validateBooleanOpt('selectBars', options.selectBars);
+  validateBooleanOpt('fixedScale', options.fixedScale);
+
+  // Validate number options
+  validateNumberOpt('labelsWidth', options.labelsWidth);
+  validateNumberOpt('tickDuration', options.tickDuration);
+  validateNumberOpt('topN', options.topN);
+  validateNumberOpt('minHeight', options.minHeight);
+  validateNumberOpt('minWidth', options.minWidth);
+  validateNumberOpt('marginTop', options.marginTop);
+  validateNumberOpt('marginRight', options.marginRight);
+  validateNumberOpt('marginBottom', options.marginBottom);
+  validateNumberOpt('marginLeft', options.marginLeft);
+
+  // Validate string options
+  validateStringOpt('theme', options.theme);
+  validateStringOpt('startDate', options.startDate);
+  validateStringOpt('endDate', options.endDate);
+
+  // Validate string or number options
+  validateStringOrNumberOpt('colorSeed', options.colorSeed);
+  validateStringOrNumberOpt('inputHeight', options.inputHeight);
+  validateStringOrNumberOpt('inputWidth', options.inputWidth);
+  validateStringOrNumberOpt('height', options.height);
+  validateStringOrNumberOpt('width', options.width);
+
+  // Validate one of options
+  const validDataShapes = ['long', 'wide'];
+  validateOneOfOpt('dataShape', options.dataShape, validDataShapes);
+
+  const validDataTypes = ['json', 'csv', 'tsv', 'xml'];
+  validateOneOfOpt('dataType', options.dataType, validDataTypes);
+
+  const validLabelsPositions = ['inside', 'outside'];
+  validateOneOfOpt('labelsPosition', options.labelsPosition, validLabelsPositions);
+
+  const validControlButtons = ['all', 'play', 'none'];
+  validateOneOfOpt('controlButtons', options.controlButtons, validControlButtons);
+
+  const validOverlays = ['all', 'play', 'repeat', 'none'];
+  validateOneOfOpt('overlays', options.overlays, validOverlays);
+
+  const validFillDateGapsIntervals = [null, 'year', 'month', 'day'];
+  validateOneOfOpt(
+    'fillDateGapsInterval',
+    options.fillDateGapsInterval,
+    validFillDateGapsIntervals,
+  );
+
+  const validFillDateGapsValues = ['last', 'interpolate'];
+  validateOneOfOpt('fillDateGapsValue', options.fillDateGapsValue, validFillDateGapsValues);
+
+  // Validate array of options
+  validateIsArrayOfType('fixedOrder', options.fixedOrder, 'string');
+
+  // Validate function or string options
+  validateFuncOrStringOpt('title', options.title);
+  validateFuncOrStringOpt('subTitle', options.subTitle);
+  validateFuncOrStringOpt('dateCounter', options.dateCounter);
+  validateFuncOrStringOpt('caption', options.caption);
+
+  // Other validations
+  validateColorMap(options.colorMap);
+  validateDataTransform(options.dataTransform);
+}
+
+function validateBooleanOpt(label: string, value: boolean | undefined): void {
+  if (typeof value !== 'undefined' && typeof value !== 'boolean') {
+    throw new Error(`Invalid ${label}: ${value}. Must be a boolean.`);
```

Comment:
Let's not throw with invalid options, specially that we are not handling these errors.
Invalid options should just be discarded.
However, you may `console.warn` to give a hint to the user if you wish. And even better, wrap it in a `reportError` function that we can enable or disable.

I would only throw errors if we are not able to render the chart at all (e.g. if we are [not able to load the data](https://github.com/hatemhosny/racing-bars/blob/6312407e7784583fe676440afa0a27979acde11e/src/lib/load-data.ts#L23)). On the other hand, all the properties in the `options` configuration object are optional. The script should not crash because the user misspelled a value.